internal/pkg/translator/obj2cil.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package translator
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	selxv1alpha2 "sigs.k8s.io/security-profiles-operator/api/selinuxprofile/v1alpha2"
+)
+
+func Object2CIL(
+	systemInherits []string,
+	objInherits []selxv1alpha2.SelinuxProfileObject,
+	sp *selxv1alpha2.SelinuxProfile,
+) string {
+	cilbuilder := strings.Builder{}
+	cilbuilder.WriteString(getCILStart(sp))
+	for _, inherit := range systemInherits {
+		cilbuilder.WriteString(getCILInheritline(inherit))
+	}
+	for _, inherit := range objInherits {
+		cilbuilder.WriteString(getCILInheritline(inherit.GetPolicyUsage()))
+	}
+	for ttype, tclassMap := range sp.Spec.Allow {
+		for tclass, perms := range tclassMap {
+			cilbuilder.WriteString(getCILAllowLine(sp, ttype, tclass, perms))
+		}
+	}
+	cilbuilder.WriteString(getCILEnd())
+	return cilbuilder.String()
+}
```

Comment:
Any scenarios in which the plain-text input (name, namespace, ref policy) could be used to impact the resulting policy? Do we need any validation here to enforce that?

Reply:
@pjbgf that's a great point! The name and namespace have a limited character set, so we're safe in the sense that no policy injection or manipulation could be done there.

Inherits coming from another SelinuxProfile are also safe because that would be the object's name. Even if one would introduce a malformed name here, the operator would try to fetch the object beforehand and fail the validation, thus not getting here at all.

System objects could be validated better, as it seems we would accept whatever we put in the SPOD object itself.

Related to types, classes and allowed actions, we could definitely validate those better. As a subsequent PR, I'd like to get the `Validate()` function to explicitly verify that the action is being allowed on an existing label, else it should fail.